### PR TITLE
refactor/2238 attestation validation and latest API

### PR DESCRIPTION
**Pull request by eoln**
- fido-credential utility module with attestation validation implementation for webauthn protocol
- use latest api-snippets
- fix integration/unit tests
TODO: get the agreement from the team & designers which part of FIDO credential should be persisted in the database

### CODE REVIEW

File: src/domain/fido-credential.ts [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+  attestationObject: string,
+  clientDataJSON: string
+): Promise<boolean> {
+  const attestation = await unpackAttestationObject(attestationObject)
+  const { attStmt, authData } = attestation
+
+  const auth = parseRegCredAuthData(Buffer.from(authData))
+
+  // user must be presented!
+  if (!(auth.flags & U2F_USER_PRESENTED)) {
```
Comment (lewisdaly) [inline src/domain/fido-credential.ts:236]: Did you mean `&&` here? Or bitwise `AND`?

File: test/unit/domain/fido-credential.test.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+ - Paweł Marzec <pawel.marzec@modusbox.com>
+ --------------
+ ******/
+import { thirdparty as tpAPI } from '@mojaloop/api-snippets'
+import { validate } from '~/domain/fido-credential'
+
+describe('fido-credential', () => {
+  describe('validate', () => {
+    it('should accept valid credential', async () => {
```
Comment (lewisdaly) [inline test/unit/domain/fido-credential.test.ts:34]: Can you also add test cases for invalid credentials? Even just 2 to start with:
1. `type !== 'public-key'`
2. Unsupported attestation format

File: src/domain/fido-credential.ts (hunk 1 above)
Comment (eoln) [reply src/domain/fido-credential.ts:236]: it is a bit flag so bitwise AND

File: test/unit/domain/fido-credential.test.ts (hunk 2 above)
Comment (eoln) [reply test/unit/domain/fido-credential.test.ts:34]: will do

**Comment by kleyow** [comment]
https://www.npmjs.com/package/fido2-lib this fido 2.0/webauthn library looks promising @eoln

**Review by kleyow — APPROVED** [review]
lgtm

File: test/unit/domain/fido-credential.test.ts (hunk 2 above)
Comment (eoln) [reply test/unit/domain/fido-credential.test.ts:108]: tests done

File: src/domain/fido-credential.ts (hunk 1 above)
Comment (eoln) [reply src/domain/fido-credential.ts:241]: it must be bitwise

**Comment by kleyow** [comment]
@eoln Whats the status on this? I've been working on a PR that will bring auth-service up to speed. Updating inferfaces, types, packages, removing outdated endpoints.

Would like this to be checked in and any update you have on current in-progress work so I get an idea if I might have conflicts.
Know you are busy! Let me get all the annoying stuff updated for you.

auth-service sure is far behind...my initial PR might be a bloodbath of old code.

**Comment by eoln** [comment]
@kleyow it was ready to merge :-)

**Pull request merged**